src/functions.js

```diff
@@ -0,0 +1,30 @@
+function addTask(tasks, input) {
+  if (input.value.trim() === '') return;
+
+  tasks.push({
+    description: input.value,
+    completed: false,
+    index: tasks.length,
+  });
+
+  input.value = '';
+
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function updateTask(tasks, task) {
+  tasks[task.index - 1] = task;
+
+  document
+    .querySelectorAll('li')
+    .forEach((elm) => elm.classList.remove('active'));
+
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function deleteTask(tasks, index) {
+  tasks = tasks.filter((t) => t.index !== index);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
```

Comment:
- [ ] when deleting a task it is not updating the localStorage/Task array correctly and therefore when deleting a task and after that adding a task it is adding the deleted task as well see below

https://user-images.githubusercontent.com/39240820/149138747-0ed2191c-deeb-4e91-bee9-7e207c8ab248.mp4